src/model/page-config-resolver.js

```diff
@@ -292,6 +335,120 @@ class MicrodataParser {
     this.doc_ = doc;
   }
 
+  /**
+   * Returns true if access is restricted, otherwise false
+   * @param {Node} root A node that is an item of type 'NewsArticle'
+   * @return {?boolean} locked access
+   * @private
+   */
+  discoverLocked_(root) {
+    const ALREADY_SEEN = 'alreadySeenForAccessInfo';
+    const nodeList = root
+        .querySelectorAll("[itemprop='isAccessibleForFree']");
+    for (let i = 0; nodeList[i]; i++) {
+      const element = nodeList[i];
+      const content = element.content || element.textContent;
+      if (!content) {
+        continue;
+      }
+      if (content.toLowerCase() == 'true') {
+        value = true;
+      } else if (content.toLowerCase() == 'false') {
+        value = false;
+      }
+      if (this.isValidElement_(element, root, ALREADY_SEEN)) {
+        return !value;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Verifies if a node is valid based on the following
+   * - child of an item of type 'NewsArticle'
+   * - not a child of an item of type 'Section'
```

Comment:
As described before - this is not correct. It should really be "not a child of any other itemtype, including WebPageElement". Confusion probably happened b/c I mistakenly called `WebPageElement` as a "section".

Reply:
Updated.